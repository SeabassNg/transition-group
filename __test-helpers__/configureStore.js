import { createStore } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly'


const reducer = (state = { num: 0, empty: false }, action = {}) => {
  switch (action.type) {
    case 'SWITCH':
      return {
        ...state,
        num: state.num + 1,
      }
    case 'EMPTY':
      return {
        ...state,
        empty: true,
      }
    case 'FULL':
      return {
        ...state,
        empty: false,
      }
    default:
      return state
  }
}

export default (initialAction) => {
  const enhancer = composeWithDevTools({ name: 'AnimatedTransitionGroup' })
  const store = createStore(reducer, enhancer())

  if (initialAction) {
    store.dispatch(initialAction)
  }

  return store
}

