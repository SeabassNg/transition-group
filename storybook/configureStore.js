import { createStore, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly'


const reducer = (state = { num: 0 }, action = {}) => {
  switch (action.type) {
    case 'SWITCH':
      return {
        num: state.num + 1,
      }
    default:
      return state
  }
}

export default () => {
  const middlewares = applyMiddleware(thunk)
  const composeEnhancers = composeWithDevTools({ name: 'AnimatedTransitionGroup' })

  return createStore(reducer, composeEnhancers(middlewares))
}

