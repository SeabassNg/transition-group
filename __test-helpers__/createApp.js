import React from 'react'
import { connect } from 'react-redux'
import { createApp } from 'jest-redux-snap'

import configureStore from './configureStore'
import { TransitionGroup, Transition } from '../src/index'

import './styles.css'


export default (groupProps, childProps, initialAction) => {
  const store = configureStore(initialAction)
  const TransitionGroup = transitionGroupHOC(groupProps, childProps)
  return createApp(store, TransitionGroup)
}

export const transitionGroupHOC = (groupProps, childProps) => {
  const Wrapper = ({ num, empty }) =>
    !empty ?
      <TransitionGroup component="div" {...groupProps}>
        <Transition key={num} {...childProps}>
          <div className='child' style={{ backgroundColor: color(num) }} />
        </Transition>
      </TransitionGroup>
      : <TransitionGroup component="div" {...groupProps} />

  return connect(state => state)(Wrapper)
}

const color = num =>
  num % 2 === 0 ? 'blue' : 'purple'

