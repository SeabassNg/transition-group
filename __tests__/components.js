import createApp from '../__test-helpers__/createApp'
import configureStore from '../__test-helpers__/configureStore'


jest.mock('../src/dom-utils', () => ({
  addClass: (component, className, name) => {
    const log = { method: 'addClass', className, name }
    console.log(log)
    expect(log).toMatchSnapshot()
  },
  removeAnimationClasses: (component, name) => {
    const log = { method: 'removeAnimationClasses', name }
    console.log(log)
    expect(log).toMatchSnapshot()
  },
  setTimeoutAnimationFrame: (func, ms, className) => {
    const log = { method: 'setTimeoutAnimationFrame', ms, className }
    console.log(log)
    expect(log).toMatchSnapshot()

    func()
  },
}))


describe('<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props', () => {
  it('appear', () => {
    const app = createApp()

    console.log(app.tree())
    console.log(app.tree().children[0])

    app.snap()

    return app.story()
  })

  it('enter + leave', () => {
    const app = createApp()

    app.snap({ type: 'SWITCH' })

    console.log(app.tree())
    console.log(app.tree().children[0])

    return app.story()
  })

  it('enter + leave -- simple duration', () => {
    const app = createApp({ duration: 500 })

    app.snap({ type: 'SWITCH' })

    console.log(app.tree())
    console.log(app.tree().children[0])

    return app.story()
  })
})


describe('<AnimatedTransitionGroup /> -- all props filled', () => {
  it('appear', () => {
    const onAppear = jest.fn()
    const app = createApp(allProps({ onAppear }))

    console.log(app.tree())
    console.log(app.tree().children[0])

    expect(onAppear).toBeCalled()

    app.snap()

    return app.story()
  })

  it('enter', () => {
    const onEnter = jest.fn()
    const app = createApp(allProps({ onEnter }), {}, { type: 'EMPTY' })

    app.snap({ type: 'FULL' })

    console.log(app.tree())
    console.log(app.tree().children[0])
    expect(onEnter).toBeCalled()

    return app.story()
  })

  it('leave', () => {
    const onLeave = jest.fn()
    const app = createApp(allProps({ onLeave }))

    app.snap({ type: 'EMPTY' })

    console.log(app.tree())
    expect(onLeave).toBeCalled()

    return app.story()
  })

  it('onFull callback', () => {
    const onFull = jest.fn()
    const app = createApp({ onFull }, {}, { type: 'EMPTY' })

    console.log(app.tree())
    app.snap({ type: 'FULL' })

    console.log(app.tree())
    expect(onFull).toBeCalled()

    return app.story()
  })

  it('onEmpty callback', () => {
    const onEmpty = jest.fn()
    const app = createApp({ onEmpty })

    console.log(app.tree())
    app.snap({ type: 'EMPTY' })

    console.log(app.tree())
    expect(onEmpty).toBeCalled()

    return app.story()
  })
})


describe('<AnimatedChild /> -- all props filled', () => {
  it('appear', () => {
    const onAppear = jest.fn()
    const app = createApp({}, allProps({ onAppear }))

    console.log(app.tree())
    console.log(app.tree().children[0])

    expect(onAppear).toBeCalled()

    app.snap()

    return app.story()
  })

  it('enter', () => {
    const onEnter = jest.fn()
    const app = createApp({}, allProps({ onEnter }), { type: 'EMPTY' })

    app.snap({ type: 'FULL' })

    console.log(app.tree())
    console.log(app.tree().children[0])
    expect(onEnter).toBeCalled()

    return app.story()
  })

  it('leave', () => {
    const onLeave = jest.fn()
    const app = createApp({}, allProps({ onLeave }))

    app.snap({ type: 'EMPTY' })

    console.log(app.tree())
    expect(onLeave).toBeCalled()

    return app.story()
  })
})


const allProps = moreProps => ({
  prefix: 'horizontal',
  duration: 500,
  delay: 5,

  appear: 'APPEAR',
  enter: 'ENTER',
  leave: 'LEAVE',

  appearDuration: 666,
  enterDuration: 667,
  leaveDuration: 668,

  appearDelay: 10,
  enterDelay: 20,
  leaveDelay: 30,

  ...moreProps,
})
