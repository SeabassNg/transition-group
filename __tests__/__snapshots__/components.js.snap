// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnimatedChild /> -- all props filled appear 1`] = `
Object {
  "className": "APPEAR",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<AnimatedChild /> -- all props filled appear 2`] = `
Object {
  "className": "APPEAR",
  "method": "setTimeoutAnimationFrame",
  "ms": 44,
}
`;

exports[`<AnimatedChild /> -- all props filled appear 3`] = `
Object {
  "className": "APPEAR-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<AnimatedChild /> -- all props filled appear 4`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 711,
}
`;

exports[`<AnimatedChild /> -- all props filled appear 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<AnimatedChild /> -- all props filled appear 6`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
</div>
`;

exports[`<AnimatedChild /> -- all props filled enter 1`] = `
Object {
  "type": "FULL",
}
`;

exports[`<AnimatedChild /> -- all props filled enter 2`] = `
Object {
  "className": "ENTER",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<AnimatedChild /> -- all props filled enter 3`] = `
Object {
  "className": "ENTER",
  "method": "setTimeoutAnimationFrame",
  "ms": 54,
}
`;

exports[`<AnimatedChild /> -- all props filled enter 4`] = `
Object {
  "className": "ENTER-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<AnimatedChild /> -- all props filled enter 5`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 722,
}
`;

exports[`<AnimatedChild /> -- all props filled enter 6`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<AnimatedChild /> -- all props filled enter 7`] = `
Object {
  "empty": false,
  "num": 0,
}
`;

exports[`<AnimatedChild /> -- all props filled enter 8`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
</div>
`;

exports[`<AnimatedChild /> -- all props filled leave 1`] = `
Object {
  "className": "APPEAR",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<AnimatedChild /> -- all props filled leave 2`] = `
Object {
  "className": "APPEAR",
  "method": "setTimeoutAnimationFrame",
  "ms": 44,
}
`;

exports[`<AnimatedChild /> -- all props filled leave 3`] = `
Object {
  "className": "APPEAR-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<AnimatedChild /> -- all props filled leave 4`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 711,
}
`;

exports[`<AnimatedChild /> -- all props filled leave 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<AnimatedChild /> -- all props filled leave 6`] = `
Object {
  "type": "EMPTY",
}
`;

exports[`<AnimatedChild /> -- all props filled leave 7`] = `
Object {
  "className": "LEAVE",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<AnimatedChild /> -- all props filled leave 8`] = `
Object {
  "className": "LEAVE",
  "method": "setTimeoutAnimationFrame",
  "ms": 64,
}
`;

exports[`<AnimatedChild /> -- all props filled leave 9`] = `
Object {
  "className": "LEAVE-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<AnimatedChild /> -- all props filled leave 10`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 733,
}
`;

exports[`<AnimatedChild /> -- all props filled leave 11`] = `
Object {
  "empty": true,
  "num": 0,
}
`;

exports[`<AnimatedChild /> -- all props filled leave 12`] = `<div />`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props appear 1`] = `
Object {
  "className": "appear",
  "method": "addClass",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props appear 2`] = `
Object {
  "className": "appear",
  "method": "setTimeoutAnimationFrame",
  "ms": 34,
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props appear 3`] = `
Object {
  "className": "appear-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props appear 4`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 35,
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props appear 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props appear 6`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
</div>
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave -- simple duration 1`] = `
Object {
  "className": "appear",
  "method": "addClass",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave -- simple duration 2`] = `
Object {
  "className": "appear",
  "method": "setTimeoutAnimationFrame",
  "ms": 34,
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave -- simple duration 3`] = `
Object {
  "className": "appear-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave -- simple duration 4`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 535,
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave -- simple duration 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave -- simple duration 6`] = `
Object {
  "type": "SWITCH",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave -- simple duration 7`] = `
Object {
  "className": "enter",
  "method": "addClass",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave -- simple duration 8`] = `
Object {
  "className": "enter",
  "method": "setTimeoutAnimationFrame",
  "ms": 34,
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave -- simple duration 9`] = `
Object {
  "className": "enter-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave -- simple duration 10`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 535,
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave -- simple duration 11`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave -- simple duration 12`] = `
Object {
  "className": "leave",
  "method": "addClass",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave -- simple duration 13`] = `
Object {
  "className": "leave",
  "method": "setTimeoutAnimationFrame",
  "ms": 34,
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave -- simple duration 14`] = `
Object {
  "className": "leave-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave -- simple duration 15`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 535,
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave -- simple duration 16`] = `
Object {
  "empty": false,
  "num": 1,
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave -- simple duration 17`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "purple",
      }
    }
  />
</div>
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave 1`] = `
Object {
  "className": "appear",
  "method": "addClass",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave 2`] = `
Object {
  "className": "appear",
  "method": "setTimeoutAnimationFrame",
  "ms": 34,
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave 3`] = `
Object {
  "className": "appear-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave 4`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 35,
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave 6`] = `
Object {
  "type": "SWITCH",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave 7`] = `
Object {
  "className": "enter",
  "method": "addClass",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave 8`] = `
Object {
  "className": "enter",
  "method": "setTimeoutAnimationFrame",
  "ms": 34,
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave 9`] = `
Object {
  "className": "enter-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave 10`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 35,
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave 11`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave 12`] = `
Object {
  "className": "leave",
  "method": "addClass",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave 13`] = `
Object {
  "className": "leave",
  "method": "setTimeoutAnimationFrame",
  "ms": 34,
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave 14`] = `
Object {
  "className": "leave-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave 15`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 35,
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave 16`] = `
Object {
  "empty": false,
  "num": 1,
}
`;

exports[`<AnimatedTransitionGroup /> + <AnimatedChild /> -- no props enter + leave 17`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "purple",
      }
    }
  />
</div>
`;

exports[`<AnimatedTransitionGroup /> -- all props filled appear 1`] = `
Object {
  "className": "APPEAR",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled appear 2`] = `
Object {
  "className": "APPEAR",
  "method": "setTimeoutAnimationFrame",
  "ms": 44,
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled appear 3`] = `
Object {
  "className": "APPEAR-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled appear 4`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 711,
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled appear 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled appear 6`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
</div>
`;

exports[`<AnimatedTransitionGroup /> -- all props filled enter 1`] = `
Object {
  "type": "FULL",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled enter 2`] = `
Object {
  "className": "ENTER",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled enter 3`] = `
Object {
  "className": "ENTER",
  "method": "setTimeoutAnimationFrame",
  "ms": 54,
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled enter 4`] = `
Object {
  "className": "ENTER-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled enter 5`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 722,
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled enter 6`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled enter 7`] = `
Object {
  "empty": false,
  "num": 0,
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled enter 8`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
</div>
`;

exports[`<AnimatedTransitionGroup /> -- all props filled leave 1`] = `
Object {
  "className": "APPEAR",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled leave 2`] = `
Object {
  "className": "APPEAR",
  "method": "setTimeoutAnimationFrame",
  "ms": 44,
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled leave 3`] = `
Object {
  "className": "APPEAR-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled leave 4`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 711,
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled leave 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "horizontal",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled leave 6`] = `
Object {
  "type": "EMPTY",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled leave 7`] = `
Object {
  "className": "LEAVE",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled leave 8`] = `
Object {
  "className": "LEAVE",
  "method": "setTimeoutAnimationFrame",
  "ms": 64,
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled leave 9`] = `
Object {
  "className": "LEAVE-active",
  "method": "addClass",
  "name": "horizontal",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled leave 10`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 733,
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled leave 11`] = `
Object {
  "empty": true,
  "num": 0,
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled leave 12`] = `<div />`;

exports[`<AnimatedTransitionGroup /> -- all props filled onEmpty callback 1`] = `
Object {
  "className": "appear",
  "method": "addClass",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled onEmpty callback 2`] = `
Object {
  "className": "appear",
  "method": "setTimeoutAnimationFrame",
  "ms": 34,
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled onEmpty callback 3`] = `
Object {
  "className": "appear-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled onEmpty callback 4`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 35,
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled onEmpty callback 5`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled onEmpty callback 6`] = `
Object {
  "type": "EMPTY",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled onEmpty callback 7`] = `
Object {
  "className": undefined,
  "method": "setTimeoutAnimationFrame",
  "ms": 0,
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled onEmpty callback 8`] = `
Object {
  "className": "leave",
  "method": "addClass",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled onEmpty callback 9`] = `
Object {
  "className": "leave",
  "method": "setTimeoutAnimationFrame",
  "ms": 34,
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled onEmpty callback 10`] = `
Object {
  "className": "leave-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled onEmpty callback 11`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 35,
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled onEmpty callback 12`] = `
Object {
  "empty": true,
  "num": 0,
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled onEmpty callback 13`] = `<div />`;

exports[`<AnimatedTransitionGroup /> -- all props filled onFull callback 1`] = `
Object {
  "type": "FULL",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled onFull callback 2`] = `
Object {
  "className": undefined,
  "method": "setTimeoutAnimationFrame",
  "ms": 0,
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled onFull callback 3`] = `
Object {
  "className": "enter",
  "method": "addClass",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled onFull callback 4`] = `
Object {
  "className": "enter",
  "method": "setTimeoutAnimationFrame",
  "ms": 34,
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled onFull callback 5`] = `
Object {
  "className": "enter-active",
  "method": "addClass",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled onFull callback 6`] = `
Object {
  "className": "done callbacks",
  "method": "setTimeoutAnimationFrame",
  "ms": 35,
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled onFull callback 7`] = `
Object {
  "method": "removeAnimationClasses",
  "name": "",
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled onFull callback 8`] = `
Object {
  "empty": false,
  "num": 0,
}
`;

exports[`<AnimatedTransitionGroup /> -- all props filled onFull callback 9`] = `
<div>
  <div
    className="child"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
</div>
`;
